### ISSUE TITLE
[Feature Request] QoL Suggestions

### ISSUE THREAD

**Issue by rex-a-licious** [avivbeeri/kintsugi#11]
I'm so happy this mod exists! It's really well thought out, integrates with vanilla Minecraft seamlessly, and addresses some long-standing issues that I've had with the vanilla enchanting system. That said, I have some suggestions that could potentially make it even better.

- [Enchantment Descriptions](https://modrinth.com/mod/enchantment-descriptions) support for the books of knowledge and even the enchantment table itself when you hover over an enchantment.
- Ability to search for enchantments within the UI.
- Ability to upgrade enchantments on already enchanted items in the enchantment table.

I would also like to echo support of a previous feature request to add an option to use regular enchanted books in place of the books of knowledge. I think this could ease the burden of adding individual mod support for loot tables when using this mod in conjunction with others that add enchantments and structures.

Thanks for the amazing work done so far, and I look forward to the future development of this mod!

**Comment by avivbeeri** [comment]
Thanks so much for your interest and comments, I'm so glad you like it!

> [Enchantment Descriptions](https://modrinth.com/mod/enchantment-descriptions) support for the books of knowledge and even the enchantment table itself when you hover over an enchantment.

I'll have to look into how this mod works in order to support this. If you're familiar with it, I'd accept a PR.

> Ability to search for enchantments within the UI.

This could be useful, but I do worry it might take away from the vanilla-feel of the mod. Most Minecraft UI's don't include a search field, so it might seem out of place.

> Ability to upgrade enchantments on already enchanted items in the enchantment table.

Kintsugi actually  allows you to upgrade the enchantments on an already-enchanted item, by paying the difference in Lapis and XP (I think there is a slight premium, but it means you can keep a tool around for a lot longer during the game).

> I would also like to echo support of a previous feature request to add an option to use regular enchanted books in place of the books of knowledge.

I'm not sure exactly what you mean by this, could you elaborate?

**Comment by rex-a-licious** [comment]
> I'll have to look into how this mod works in order to support this. If you're familiar with it, I'd accept a PR.

I'm unfortunately unfamiliar with modding but to my knowledge, the mod uses lang files to add descriptions to enchantments.

> This could be useful, but I do worry it might take away from the vanilla-feel of the mod. Most Minecraft UI's don't include a search field, so it might seem out of place.

That's totally fair. I only suggested it because in a modded setting the list could get quite long with many enchantments added to the game, but it's not a pressing issue. Maybe it could be toggleable?

> Kintsugi actually allows you to upgrade the enchantments on an already-enchanted item, by paying the difference in Lapis and XP (I think there is a slight premium, but it means you can keep a tool around for a lot longer during the game).

Oh that's great news, I must have overlooked this feature. My apologies!

> I'm not sure exactly what you mean by this, could you elaborate?

Same suggestion as this previous request: https://github.com/avivbeeri/kintsugi/issues/6

I would like the ability to toggle off the books of knowledge and use regular enchanted books to add enchantments to the table instead. You could perhaps limit the highest enchantment level to the level of the book inside of the bookshelf. My reason for wanting this is that when using Kintsugi with other mods that add structures and enchantments, there is no way to obtain the non-vanilla books of knowledge in these new structures without the assistance of datapacks. In my modpack, I've also removed the anvil's ability to apply enchantments via the similar [Enchant & Repair](https://www.curseforge.com/minecraft/mc-mods/enchant-repair) mod so being able to use regular enchanted books instead would be especially useful to me.

Whether you implement none or all of these features, thank you for taking the time to hear me out!

**Comment by AngelOfCreation** [comment]
Would like to add my 2 cents: JEI support for knowledge tomes so you can see what items an knowledge tome can be applied to.

**Comment by avivbeeri** [comment]
> Would like to add my 2 cents: JEI support for knowledge tomes so you can see what items an knowledge tome can be applied to.

As far as I can remember, you can't apply Knowledge Tomes to anything, they only serve to go into bookshelves for the Enchanting table, so this doesn't make sense to me on first reading.

**Comment by avivbeeri** [comment]
It looks like the tooltips request is possible, based on the answer I received in the mentioned issue, so I may pursue that soon.

**Comment by AngelOfCreation** [comment]
> > Would like to add my 2 cents: JEI support for knowledge tomes so you can see what items an knowledge tome can be applied to.
> 
> As far as I can remember, you can't apply Knowledge Tomes to anything, they only serve to go into bookshelves for the Enchanting table, so this doesn't make sense to me on first reading.

Yes, but their corresponding enchantments book *can* go on items. So you could integrate with JEI and EMI to take the data from the enchantment books about what items an enchantment can be applied to and create a new tab that show the same data but for knowledge tomes. That way you don't have to search in JEI/EMI for the corresponding enchantment book before being able to look at its item applications.
